plugin/storage/memory/memory.go

```diff
@@ -34,7 +34,7 @@ type Store struct {
 	ids        []*model.TraceID
 	traces     map[model.TraceID]*model.Trace
 	services   map[string]struct{}
-	operations map[string]map[string]struct{}
+	operations map[string]map[string]map[string]struct{}
```

Comment:
I find it very hard to reason about 3 nested maps. IIRC the `services` is just a set of services, and `operations` is a map of service -> operations set. In the new implementation you want it to be a map of service -> operation+spanKind. So can it be something like `map[string]map[Operation]struct{}`?

Reply:
Updated